src/Umbraco.Infrastructure/PropertyEditors/TagsPropertyEditor.cs

```diff
@@ -146,46 +167,45 @@ public IEnumerable<ITag> GetTags(object? value, object? dataTypeConfiguration, i
                     throw new NotSupportedException($"Value \"{tagConfiguration.StorageType}\" is not a valid TagsStorageType.");
             }
 
-            return tags.Select(x => new Tag
-            {
-                Group = tagConfiguration.Group,
-                Text = x,
-                LanguageId = languageId,
-            });
+            return tags.ToArray();
         }
 
-
         /// <inheritdoc />
         public override IValueRequiredValidator RequiredValidator => new RequiredJsonValueValidator();
 
-        /// <inheritdoc />
-        public override object? FromEditor(ContentPropertyData editorValue, object? currentValue)
+        public override object? ToEditor(IProperty property, string? culture = null, string? segment = null)
         {
-            var value = editorValue.Value?.ToString();
-
-            if (string.IsNullOrEmpty(value))
+            var val = property.GetValue(culture, segment);
+            if (val == null)
```

Comment:
More pattern matching 😁 
```suggestion
            if (val is null)
```